network/message_codes.go

```diff
@@ -0,0 +1,354 @@
+package network
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/libp2p/message"
+	"github.com/onflow/flow-go/model/messages"
+	"github.com/onflow/flow-go/network/codec"
+)
+
+// codeToMessageCodes is a mapping of Code -> MessageCode wrapper that contains authorization information used for validation
+type codeToMessageCodes map[codec.Code]MessageCode
+
+// MessageCode wrapper around Code that contains channel authorization information for the message Code.
+type MessageCode struct {
+	// Code is the underlying message code byte
+	Code codec.Code
+
+	// authorizedChannels full list of network.Channel this message is authorized to be sent on including channel aliases
+	authorizedChannels ChannelList
+
+	// authorizedChannelsMap is a mapping of channels to authorized roles allowed to send MessageCode on the channel
+	authorizedChannelsMap map[Channel]flow.RoleList
+}
+
+// IsAuthorizedChannel returns true if channel is in the authorizedChannels list for the message code
+func (mc MessageCode) IsAuthorizedChannel(channel Channel) bool {
+	return mc.authorizedChannels.Contains(channel)
+}
+
+// AuthorizedRolesByChannel returns the list of roles authorized to send this message code on the channel
+func (mc MessageCode) AuthorizedRolesByChannel(channel Channel) flow.RoleList {
+	return mc.authorizedChannelsMap[channel]
+}
+
+var messageCodeMap codeToMessageCodes
+
+// initializeMessageCodeMap initializes the messageCodeMap
+func initializeMessageCodeMap() {
+	messageCodeMap = make(codeToMessageCodes)
+
+	// consensus
+	messageCodeMap[codec.CodeBlockProposal] = MessageCode{
+		Code:               codec.CodeBlockProposal,
+		authorizedChannels: ChannelList{ConsensusCommittee, PushBlocks, ReceiveBlocks},
```

Comment:
authorizedChannels is kept for readability and completeness since some channels have aliases, we should explicitly list out the alias channels.